tests/foreman/cli/test_host.py

```diff
@@ -819,6 +819,41 @@ def test_positive_list_by_last_checkin(
     assert rhel7_contenthost.hostname in [host['name'] for host in hosts]
 
 
+@pytest.mark.host_create
+@pytest.mark.tier3
+def test_positive_list_infrastructure_hosts(
+    module_lce, module_org, module_promoted_cv, rhel7_contenthost, default_sat
+):
+    """List infrasturcture hosts (Satellite and Capsule)
+
+    :id: 9e4c873e-0954-4096-b337-bcd679181025
+
+    :expectedresults: Infrastructure hosts are listed
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    rhel7_contenthost.install_katello_ca(default_sat)
+    rhel7_contenthost.register_contenthost(
+        module_org.label,
+        lce=f'{module_lce.label}/{module_promoted_cv.label}',
+    )
+    assert rhel7_contenthost.subscribed
+    Host.update({'name': default_sat.hostname, 'new-organization-id': module_org.id})
+    # list satellite hosts
+    hosts = Host.list({'search': 'infrastructure_facet.foreman=true'})
+    hostnames = [host['name'] for host in hosts]
+    assert rhel7_contenthost.hostname not in hostnames
+    assert default_sat.hostname in hostnames
+    # list capsule hosts
```

Comment:
I believe this doesn't test if non-internal capsules are listed.

Reply:
true, the intention is to test the search pattern. I'm going with the internal capsule to reduce footprint, I realize it leaves a bit of a gap. However, this test is mainly a supplement to test_positive_rex_against_infra_hosts form cli rex module where the actual external capsule is used, so the theoretical external capsule issue would be caught there. Hope this makes sense